hbase-shell/src/main/ruby/shell.rb

```diff
@@ -320,10 +323,20 @@ def eval_io(io)
       scanner = RubyLex.new
       scanner.set_input(io)
 
+      scanner.each_top_level_statement do |statement, linenum|
+        puts(workspace.evaluate(nil, statement, filename, linenum))
```

Comment:
this propagates source files and line numbers for the purpose of error handling? nice!

Reply:
Yup! Credit to @busbey for figuring this out in HBASE-11658 (this code used to exist in `bin/hirb.rb`).